src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1801 @@
+        {
+            HandleCommand<Commands::CaptureSnapshot::DecodableType>(
+                handlerContext,
+                [this](HandlerContext & ctx, const auto & commandData) { HandleCaptureSnapshot(ctx, commandData); });
+        }
+        return;
+    }
+}
+
+void CameraAVStreamMgmtServer::HandleVideoStreamAllocate(HandlerContext & ctx,
+                                                         const Commands::VideoStreamAllocate::DecodableType & commandData)
+{
+    Status status = Status::Success;
+
+    Commands::VideoStreamAllocateResponse::Type response;
+    auto & streamUsage      = commandData.streamUsage;
+    auto & videoCodec       = commandData.videoCodec;
+    auto & minFrameRate     = commandData.minFrameRate;
+    auto & maxFrameRate     = commandData.maxFrameRate;
+    auto & minResolution    = commandData.minResolution;
+    auto & maxResolution    = commandData.maxResolution;
+    auto & minBitRate       = commandData.minBitRate;
+    auto & maxBitRate       = commandData.maxBitRate;
+    auto & minFragmentLen   = commandData.minFragmentLen;
+    auto & maxFragmentLen   = commandData.maxFragmentLen;
+    bool isWaterMarkEnabled = false;
+    bool isOSDEnabled       = false;
+
+    if (HasFeature(Feature::kWatermark))
+    {
+        isWaterMarkEnabled = commandData.watermarkEnabled.ValueOr(false);
+    }
+    else
+    {
+        VerifyOrExit(!commandData.watermarkEnabled.HasValue(), status = Status::InvalidCommand;
+                     ChipLogError(Zcl, "CameraAVStreamMgmt: Failed to allocate Video stream. Unsupported watermark feature"));
+    }
+
+    if (HasFeature(Feature::kOnScreenDisplay))
+    {
+        isOSDEnabled = commandData.OSDEnabled.ValueOr(false);
+    }
+    else
+    {
+        VerifyOrExit(!commandData.OSDEnabled.HasValue(), status = Status::InvalidCommand;
+                     ChipLogError(Zcl, "CameraAVStreamMgmt: Failed to allocate Video stream. Unsupported OnScreenDisplay feature"));
+    }
+
+    // Call the delegate
+    status =
+        mDelegate.VideoStreamAllocate(streamUsage, videoCodec, minFrameRate, maxFrameRate, minResolution, maxResolution, minBitRate,
+                                      maxBitRate, minFragmentLen, maxFragmentLen, isWaterMarkEnabled, isOSDEnabled);
+
+    if (status == Status::Success)
+    {
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+        return;
+    }
+
+exit:
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void CameraAVStreamMgmtServer::HandleVideoStreamModify(HandlerContext & ctx,
+                                                       const Commands::VideoStreamModify::DecodableType & commandData)
+{
+    Status status           = Status::Success;
+    bool isWaterMarkEnabled = false;
+    bool isOSDEnabled       = false;
+    auto & videoStreamID    = commandData.videoStreamID;
+
+    if (HasFeature(Feature::kWatermark))
+    {
+        isWaterMarkEnabled = commandData.watermarkEnabled.ValueOr(false);
```

Comment:
This has all the same bugs as VideoStreamAllocate in terms of allowing invalid commands, not validating enums, etc.

I'm stopping here in the .cpp.  Please fix the issues pointed out so far, including in the code I have not read yet.